conbench/entities/benchmark_result.py

```diff
@@ -426,8 +453,32 @@ class _BenchmarkFacadeSchemaCreate(marshmallow.Schema):
             )
         },
     )
-    machine_info = marshmallow.fields.Nested(MachineSchema().create, required=False)
-    cluster_info = marshmallow.fields.Nested(ClusterSchema().create, required=False)
+    machine_info = marshmallow.fields.Nested(
+        MachineSchema().create,
+        required=False,
+        metadata={
+            "description": conbench.util.dedent_rejoin(
+                """
+                Precisely one of `machine_info` and `cluster_info` must be
+                provided. The data is however ignored when the Run (referred to
+                by `run_id`) was previously created.
+                """
+            )
+        },
+    )
+    cluster_info = marshmallow.fields.Nested(
+        ClusterSchema().create,
+        required=False,
+        metadata={
+            "description": conbench.util.dedent_rejoin(
+                """
+                Precisely one of `machine_info` and `cluster_info` must be
```

Comment:
I'm not sure if this is me only, but I find the "Precisely one of ... must be provided" a unnatural sounding (it's precisely accurate, but easy to skim over that and it sounds confusing to me). At the same time, I haven't been able to come up with anything better that's not massively verbose. Happy to leave it as is, but curious if others feel the same (or have a better rephrase than I do!)

Reply:
I feel the same actually, and have always felt the same when I was in the position to describe that "exactly one of [] is required" constraint. Haha, this is a recurring problem all my programming life. I leave this as-is for now, but am also very open to more elegant, user-friendly wording.

Reply:
I'm a fan of "Either A or B", which in my English-speaking experience implies "A xor B", but I realize that it may not imply that for other people.